_releases/2023-01-09-1.7.0-released.md

```diff
@@ -0,0 +1,109 @@
+
+# Interpreter REPL
+
+The REPL interface of the interpreter has received a major usability improvement.
+The command prompt now supports multiline input, command history, and several key bindings.
+
+[![asciicast](/assets/blog/2023-01-09-interpreter-repl-asciicast.gif)](https://asciinema.org/a/oG55o2Jm3XuNj8DJKBb9u0XuW)
+
+This great feature set is provided by the shard [REPLy](https://github.com/I3oris/reply) ([#12738](https://github.com/crystal-lang/crystal/pull/12738)).
+
+# `Regex` with PCRE2
+
+Crystal relies on `libpcre` as engine for executing regular expressions in [`Regex`](https://crystal-lang.org/api/1.7.0/Regex.html).
+In 1.7.0 we're adding support for its successor, `libpcre2`, but for now it's opt-in with the compiler flag `-Duse_pcre2`.
+
+There are no major differences in usage and PCRE2 is mostly backwards compatible.
+So we don't expect many issues. But we still encourage testing out the new
+engine to see if something breaks.
+
+PCRE2 is going to be enabled implicitly in one of the next releases, likely
+becoming the default engine. PCRE will continue to serve as a fallback when the newer library is not available.
+See [#12790](https://github.com/crystal-lang/crystal/issues/12790) for the tracking issue.
+
+The regex engine is also used in the compiler for validating regex literals. Updating the compiler to PCRE2 might cause
+some friction due to minor syntactical differences. Using different regex engines in the compiler and the compiled program
+may lead to the compiler rejecting a regex literal for syntax errors, even if it's valid for the runtime engine.
+Read more about the implications on the language syntax in [#12857](https://github.com/crystal-lang/crystal/issues/12857).
+
+# Misc
+
+## Language
+
+* Lib functions are now visible in top-level macros ([#12848](https://github.com/crystal-lang/crystal/pull/12848)).
+* Added [`ArrayLiteral#-(other)`](https://crystal-lang.org/api/1.7.0/Crystal/Macros/ArrayLiteral.html#-(other:ArrayLiteral):ArrayLiteral-instance-method) and [`TupleLiteral#-(other)`](https://crystal-lang.org/api/1.7.0/Crystal/Macros/TupleLiteral.html#-(other:TupleLiteral):TupleLiteral-instance-method) ([#12646](https://github.com/crystal-lang/crystal/pull/12646)).
+* `annotation` supports the [`@[Deprecated]`](https://crystal-lang.org/api/1.7.0/Deprecated.html) annotation now ([#12557](https://github.com/crystal-lang/crystal/pull/12557)).
+* Removed oct/bin floating point literals. This number syntax had been broken and isn't very useful anyway... ([#12687](https://github.com/crystal-lang/crystal/pull/12687)).
+
+## Compiler
+
+Clear error message when declaring a constant within another constant declaration ([#12566](https://github.com/crystal-lang/crystal/pull/12566)).
+
+```cr
+A = B = 1 # Syntax Error: dynamic constant assignment
+```
+
+Fixed wrong/missing locations of various AST nodes in the parser ([#11798](https://github.com/crystal-lang/crystal/pull/11798)).
```

Comment:
Are these important to stand out here?

```suggestion
```

Reply:
Not really, they're on my maybe list.
Happy to drop it.